### ISSUE TITLE
Error when compiling samples in latest Nim versions (with workaround)

### ISSUE THREAD

**Issue by Skaruts** [Vladar4/libtcod_nim#16]
I got this error when compiling the samples in the latest Nim version 1.2.2:

```
C:\...\samples112\samples.nim(1489, 31) Error: type mismatch: got <tuple of (string, proc (first: bool, key: ptr Key, mouse: ptr Mouse){.cdecl, locks: <unknown>.})> but expected 'tuple of (string, proc (first: bool, key: ptr Key, mouse: ptr Mouse){.cdecl, gcsafe, locks: 0.})'
```

To make it work I added the `gcsafe` pragma to this line:
https://github.com/Vladar4/libtcod_nim/blob/8c443ae4b9ce80f4632a0ffdfd3f3a43dd893232/samples112/samples.nim#L1145

But it throws this warning, though:
```
C:\...\libtcod_nim\samples112\samples.nim(1145, 6) Warning: 'renderName' is not GC-safe as it accesses 'names' which is a global using GC'ed memory [GcUnsafe2]
```

---

By the way, there's also this warning lying around there:
```
C:\...\.nimble\pkgs\libtcod_nim-1.112.0\src112\sys.nim(161, 53) Warning: use `csize_t` instead; csize is deprecated [Deprecated]
```

**Comment by Vladar4** [comment]
Fixed in 539f7f0c8a2d1e99689fdbd7d38f1f05bb1f79d7
Please check. It still throws a warning but everything works as intended, it seems.

**Comment by Skaruts** [comment]
Yup. It works. Thanks.
